src/main/resources/avro/variants.avdl

```diff
@@ -189,20 +275,45 @@ record Variant {
   Genomic positions are non-negative integers less than reference length.
   Variants spanning the join of circular genomes are represented as
   two variants one on each side of the join (position 0).
+
+  If the API server supports the "classic" mode, this field must not be null.
   */
-  long start;
+  union { null, long } start = null;
 
   /**
   The end position (exclusive), resulting in [start, end) closed-open interval.
   This is typically calculated by `start + referenceBases.length`.
+
+  If the API server supports the "classic" mode, this field must not be null.
   */
-  long end;
+  union { null, long } end = null;
 
-  /** The reference bases for this variant. They start at the given position. */
-  string referenceBases;
+  /**
+  The reference bases for this variant. They occupy the range described by
+  `referenceName`, `start`, and `end`, which all must be set if this field is
+  set.
 
-  /** The bases that appear instead of the reference bases. */
-  array<string> alternateBases = [];
+  If the API server supports the "classic" mode, this field must not be null.
+  */
+  union { null, string } referenceBases = null;
+
+  /**
+  The bases that appear instead of the reference bases. If this field is not
+  null, `referenceBases` must be set.
+
+  If the API server supports the "classic" mode, this field must not be null.
+  */
+  union { null, array<string> } alternateBases = null;
+
+  /**
+  The IDs of the reference and alternate `Allele`s for this `Variant`. The first
+  is the reference allele, and any others are alternates. If this field is set
+  along with `referenceName`, `start`, `end`, `referenceBases`, and/or
+  `alternateBases`, those fields must agree with the `Allele`s given here.
+
+  If the API server supports the "graph" mode, this field must not be null.
+  */
+  union { null, array<string> } alleleIds;
```

Comment:
Nope. They have to be non-null if the server supports "calssic" mode, but they don't have to be null if the server supports "graph" mode, because a server can support "classic" and "graph" mode at the same time. 

You can have both groups of fields set; they just need to agree with each other (which in practice means you can't use some of the features of the graph reference).